packages/compiler-sfc/src/templateTransformAssetUrl.ts

```diff
@@ -120,12 +120,16 @@ export const transformAssetUrl: NodeTransform = (
           attr.value.content[0] !== '@' &&
           isRelativeUrl(attr.value.content)
         ) {
+          // Allow for full hostnames provided in options.base
+          const base = parseUrl(options.base)
+          const host = base.host ? base.protocol + '//' + base.host : ''
+          const basePath = base.path || options.base
+
           // when packaged in the browser, path will be using the posix-
           // only version provided by rollup-plugin-node-builtins.
-          attr.value.content = (path.posix || path).join(
-            options.base,
-            url.path + (url.hash || '')
-          )
+          attr.value.content =
+            host +
+            (path.posix || path).join(basePath, url.path + (url.hash || ''))
```

Comment:
`path.join()` does not work well with full URLs, which is why I'm prepending `host` here.

```js
// Notice how path.join() strips the second slash from http://
// This is the problem that is solved by this pull request.
path.join("http://localhost:3000/src/assets", "logo.png") === 'http:/localhost:3000/src/assets/logo.png')
```